packages/roosterjs-content-model-plugins/lib/imageEdit/ImageEditPlugin.ts

```diff
@@ -0,0 +1,461 @@
+import { applyChange } from './utils/applyChange';
+import { ChangeSource } from 'roosterjs-content-model-dom';
+import { checkIfImageWasResized } from './utils/imageEditUtils';
+import { createImageWrapper } from './utils/createImageWrapper';
+import { Cropper } from './Cropper/cropperContext';
+import { getDropAndDragHelpers } from './utils/getDropAndDragHelpers';
+import { getHTMLImageOptions } from './utils/getHTMLImageOptions';
+import { getImageEditInfo } from './utils/getImageEditInfo';
+import { ImageEditElementClass } from './types/ImageEditElementClass';
+import { RESIZE_IMAGE } from './constants/constants';
+import { Resizer } from './Resizer/resizerContext';
+import { Rotator } from './Rotator/rotatorContext';
+import { updateWrapper } from './utils/updateWrapper';
+import type { DragAndDropHelper } from '../pluginUtils/DragAndDrop/DragAndDropHelper';
+import type { DragAndDropContext } from './types/DragAndDropContext';
+import type { ImageHtmlOptions } from './types/ImageHtmlOptions';
+import type { ImageEditOptions } from './types/ImageEditOptions';
+import type {
+    EditAction,
+    EditorPlugin,
+    IEditor,
+    ImageMetadataFormat,
+    PluginEvent,
+    SelectionChangedEvent,
+} from 'roosterjs-content-model-types';
+
+const DefaultOptions: Partial<ImageEditOptions> = {
+    borderColor: '#DB626C',
+    minWidth: 10,
+    minHeight: 10,
+    preserveRatio: true,
+    disableRotate: false,
+    disableSideResize: false,
+    onSelectState: 'resizeAndRotate',
+};
+
+/**
+ * ImageEdit plugin handles the following image editing features:
+ * - Resize image
+ * - Crop image
+ * - Rotate image
+ * - Flip image
+ */
+export class ImageEditPlugin implements EditorPlugin {
```

Comment:
Implement ImageEditor interface so ribbon buttons and context menu can call it direclty, then no need to handle ImageEdit event